### ISSUE TITLE
HCOM: Crash if trying to change non-existing parameter with same name as variable

### ISSUE THREAD

**Issue by robbr48** [Hopsan/hopsan#2115]
**Describe the bug**
Program will crash if user tries to change a model parameter that does not exist, if a variable of the same name exists locally in HCOM.

**To Reproduce**
1. Open a new model without a system parameter named "x"
2. > x = 5
3. > chpa x 3
4. Program will crash

**Expected behavior**
Show some clever error message.

**Comment by robbr48** [comment]
Seems like it crashes because the code tries to evaluate the parameter name by something called "component index". I have no idea what this code is supposed to do.
https://github.com/Hopsan/hopsan/blob/master/HopsanGUI/HcomHandler.cpp#L2119
```
//If no parameters found, try if component name can be evaluated by component index instead
if(parameterNames.isEmpty())
{
    evaluateExpression(splitCmd[0].section(".",0,0));
    if(mAnsType == Scalar)
    {
        QString compName = pSystem->getModelObjects().at(mAnsScalar)->getName();
        splitCmd[0].replace(0, splitCmd[0].section(".",0,0).size(), compName);
        getParameters(splitCmd[0], parameterNames);
    }
}
```

**Comment by peterNordin** [comment]
I dont know either, get rid of it. 
Probably some ancient stuff, it is not possible for the user to expect at which index in the pointer vector a particular component resides.

**Comment by peterNordin** [comment]
A git blame shows you as the author :)

**Comment by robbr48** [comment]
Yes, I noticed :) It might have been related to my multicore benchmark models, which were autogenerated by HCOM and had very many components of the same type. But it is useless today, so I will remove it.

**Comment by robbr48** [comment]
Resolved by #2225.
